test/test_kpt_models.cpp

```diff
@@ -0,0 +1,275 @@
+#include <test/eos_unit_test_helpers.hpp>
+
+using singularity::LogMaxTimeStep;
+using singularity::LogRatesCGModel;
+using singularity::SmallStepMFUpdate;
+using singularity::SortGibbs;
+
+SCENARIO("First log rate test") {
+  GIVEN("Parameters for a 5 phase system") {
+    constexpr Real w[25] = {0.,  0.8510E+01, 20., 20., 20., 0.8510E+01, 0.,  20., 20.,
+                            20., 20.,        20., 0.,  20., 20.,        20., 20., 20.,
+                            0.,  20.,        20., 20., 20., 20.,        0.};
+
+    constexpr Real b[25] = {0.,         0.3060E-04, 0.1000E-05, 0.1000E-05, 0.1000E-05,
+                            0.3060E-04, 0.,         0.1000E-05, 0.1000E-05, 0.1000E-05,
+                            0.1000E-05, 0.1000E-05, 0.,         0.1000E-05, 0.1000E-05,
+                            0.1000E-05, 0.1000E-05, 0.1000E-05, 0.,         0.1000E-05,
+                            0.1000E-05, 0.1000E-05, 0.1000E-05, 0.1000E-05, 0.};
+
+    GIVEN("Gibbs free energy") {
+      constexpr int num = 5;
+      constexpr int mnum = num * (num - 1) / 2;
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+      // Create Kokkos views on device for the input arrays
+      // Create host-side mirrors of the inputs
+      Kokkos::View<Real[num]> v_gibbs("gibbs");
+      auto gibbs = Kokkos::create_mirror_view(v_gibbs);
+      // order phases from high to low gibbs
+      Kokkos::View<int[num]> v_order("order");
+      auto order = Kokkos::create_mirror_view(v_order);
+#else
+      // Otherwise just create arrays to contain values and create pointers to
+      // be passed to the functions in place of the Kokkos views
+      std::array<Real, num> gibbs;
+      std::array<int, num> order;
+      auto v_gibbs = gibbs.data();
+      auto v_order = order.data();
+#endif // PORTABILITY_STRATEGY_KOKKOS
+
+      // Populate the input arrays
+      portableFor(
+          "Initialize gibbs", 0, 1, PORTABLE_LAMBDA(int i) {
+            v_gibbs[0] = 0.0292971;
+            v_gibbs[1] = 0.0286937;
+            v_gibbs[2] = 0.0287409;
+            v_gibbs[3] = 1.0e99;
+            v_gibbs[4] = 0.02940076;
+          });
+#ifdef PORTABILITY_STRATEGY_KOKKOS
+      // copy the inputs
+      Kokkos::deep_copy(gibbs, v_gibbs);
+#endif // PORTABILITY_STRATEGY_KOKKOS
+
+      // Gold standard values for a subset of lookups
+      constexpr std::array<Real, mnum> logrates_true{
+          1.00000000000000007e+210, 1.00000000000000007e+210, 1.00000000000000007e+210,
+          1.00000000000000007e+210, 4.99943400447804946e+05,  4.35424707359967171e+05,
+          1.07530324485867186e+04,  3.93959978909504514e+02,  3.09367756860215100e+05,
+          2.23469012616621285e+03};
+      constexpr std::array<int, num> phaseorder{0, 1, 2, 3, 4};
+      constexpr std::array<int, num> order_true{3, 4, 0, 2, 1};
+
+      WHEN("Using SortGibbs(num,gibbs,order) to give phases in order of largest to "
+           "smallest gibbs") {
+        SortGibbs(num, v_gibbs, v_order);
```

Comment:
Once we do the GPU stuff, this test may need to change in small ways---i.e., what things are on host, what things are on device, etc. We can discuss in person.